### PR TITLE
CAMEL-19864: fix default value for Univocity parsers line separator.

### PR DESCRIPTION
**Pull request by klease**
Use \n instead of \\n and adapt the utility class used to generate SpringBoot configuration file from the json files.

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Comment by davsclaus** [comment]
Can you rebuild source and update PR as there are uncommitted files.

mvn clean install -Dquickly

**Review by oscerd — APPROVED** [review]
Lgtm. Just need a full build

**Comment by klease** [comment]
I'm on it but due to a heavy schedule, not sure if I have time to commit during the day.

**Comment by davsclaus** [comment]
No problem, one of us can likely do this and update the PR. Thanks for the status update.

**Comment by klease** [comment]
I committed the generated catalog and camel-yaml-dsl files which look OK. Unfortunately the file catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd is not correct because it previously had the correct escape sequence (single backslash + n) and now it has an actual newline. So the generator needs to be reviewed. Also I didn't check the generation of the .adoc files which might be affected since currently they also have a single backslash.

**Comment by davsclaus** [comment]
We could also remove that \n as default value, and just mention it in the docs / description.

**Comment by klease** [comment]
> We could also remove that \n as default value, and just mention it in the docs / description.
True, but it seems OK now after the last commits.

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd [hunk 1]
```diff
@@ -9382,7 +9382,7 @@ The line separator of the files The default value is to use the JVM platform lin
           <xs:annotation>
             <xs:documentation xml:lang="en">
 <![CDATA[
-The normalized line separator of the files The default value is a new line character. Default value: \n
+The normalized line separator of the files The default value is a new line character. Default value:
```
Comment (davsclaus) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd:9385]: Only here its not correct, but its something we can accept. Or you need to check the maven plugin that generates these comments: camel-eip-documentation-enricher-maven-plugin

Comment (klease) [reply catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd:9385]: Yes, true; we could do this:  text += " Default value: " + defaultValueText.toString().replace("\n", "\\n");

**Comment by klease** [comment]
I see the generated website files also have an issue where the newline appears literally. So I'm closing this pull request. Perhaps the easiest solution is what @davsclaus suggested which is to remove the default value. In fact these defaults are **not** present in the Camel Univocity component itself but in the underlying parser. Only in camel-spring-boot does the autoconfiguration set them on the Camel component itself.

**Pull request closed without merging**